### PR TITLE
Fix #14919: truncate long function names in Vv

### LINKED ISSUE #14919: Chop the long lines in `Vv` mode
When the function name is long enough it overlaps the disassembly window in `Vv` mode.
I think it should be chopped up to that limit.

### PR DESCRIPTION
**Pull request by deepakchethan**
After change:
Actual function name: sym.android_app_Activity.onCreateOptionsMenu (as seen as fcn in disasm)

Screen Shot 2019-09-09 at 11 16 33 PM

### CODE REVIEW

**Comment by radare** [comment]
there are more thngs that fuckupthis mode. we should use r_str_crop for this instead of asuming a fixed size. also i think [] keys should be able to  move the right column right/left

Screenshot 2019-09-09 at 19 59 50

**Comment by deepakchethan** [comment]
I have updated the pr, can you take a look @radare

File: libr/core/vmenus.c [hunk 1]
```diff
@@ -3134,6 +3149,10 @@ R_API void r_core_visual_anal(RCore *core, const char *input) {
 		}
 		ch = r_cons_arrow_to_hjkl (ch); // get ESC+char, return 'hjkl' char
 		switch (ch) {
+		case '[':
+			show_vars = true; break;
```
Comment (radare) [inline libr/core/vmenus.c]: Break goes into the next line

File: libr/core/vmenus.c [hunk 2]
```diff
@@ -2724,18 +2728,29 @@ static ut64 var_functions_show(RCore *core, int idx, int show) {
...
 							color_fcn, fcn->name);
 				} else {
-					r_cons_printf ("%c%c 0x%08"PFMT64x" %4d %s\n",
+					var_functions = r_str_newf ("%c%c 0x%08"PFMT64x" %4d %s",
 							(seek == fcn->addr)?'>':' ',
 							(idx==i)?'*':' ',
 							fcn->addr, r_anal_fcn_realsize (fcn), fcn->name);
 				}
+				if (!show_vars) {
+					tmp = r_str_crop (var_functions, 0, 0, 72, r_cons_singleton()->rows);
```
Comment (radare) [inline libr/core/vmenus.c]: Dont hardcode this 72. Imho it should be aligned with asm.cmt.col or whatever its used...and if its hardcoded should reuse the same const

File: libr/core/vmenus.c [hunk 3]
```diff
@@ -2724,18 +2728,29 @@ static ut64 var_functions_show(RCore *core, int idx, int show) {
...
+					if (strlen (tmp) < strlen (var_functions)) {
```
Comment (radare) [inline libr/core/vmenus.c]: Better use ansilen

File: libr/core/vmenus.c [hunk 4]
```diff
@@ -2724,18 +2728,29 @@ static ut64 var_functions_show(RCore *core, int idx, int show) {
 				addr = fcn->addr;
 			}
 			if (show) {
+				char *tmp;
 				if (color) {
-					r_cons_printf ("%c%c %s0x%08"PFMT64x"" Color_RESET" %4d %s%s"Color_RESET"\n",
+					var_functions = r_str_newf ("%c%c %s0x%08"PFMT64x"" Color_RESET" %4d %s%s"Color_RESET"",
```
Comment (radare) [inline libr/core/vmenus.c]: Varfunctions is leaking i think

File: libr/core/vmenus.c [hunk 5]
```diff
@@ -2724,18 +2728,29 @@ static ut64 var_functions_show(RCore *core, int idx, int show) {
...
 							fcn->addr, r_anal_fcn_realsize (fcn), fcn->name);
 				}
+				if (!show_vars) {
+					tmp = r_str_crop (var_functions, 0, 0, 72, r_cons_singleton()->rows);
+					if (strlen (tmp) < strlen (var_functions)) {
+						r_cons_printf ("%s..%s\n", tmp, Color_RESET);
+						print_full_func = false;
+					}
+				}
+				if (print_full_func){
```
Comment (radare) [inline libr/core/vmenus.c]: Missing space

File: libr/core/vmenus.c (hunk 4 above)
Comment (deepakchethan) [reply libr/core/vmenus.c]: Yes, My bad

File: libr/core/vmenus.c (hunk 2 above)
Comment (deepakchethan) [reply libr/core/vmenus.c]: Done

**Comment by deepakchethan** [comment]
@radare @XVilka  can you please review it

File: libr/core/vmenus.c [hunk 6]
```diff
@@ -2715,27 +2718,45 @@ static ut64 var_functions_show(RCore *core, int idx, int show) {
...
 							(idx==i)?'*':' ',
 							fcn->addr, r_anal_fcn_realsize (fcn), fcn->name);
 				}
+				if (var_functions) {
+					if (!show_vars) {
+						int fun_len = r_str_ansi_len (var_functions);
+						int columns = fun_len > cols ? cols - 2 : cols;
+						tmp = r_str_ansi_crop (var_functions, 0, 0, columns, window);
+						if (r_str_ansi_len (tmp) < fun_len) {
+							r_cons_printf("%s..%s\n", tmp, Color_RESET);
```
Comment (radare) [inline libr/core/vmenus.c]: Missing space

File: libr/core/vmenus.c [hunk 7]
```diff
@@ -2968,7 +2989,19 @@ static ut64 r_core_visual_anal_refresh (RCore *core) {
 		}
 		// TODO: filter only the callrefs. but we cant grep here
 		sprintf (old, "afi @ 0x%08"PFMT64x, addr);
-		r_core_cmd0 (core, old);
+		char *output = r_core_cmd_str (core, old);
+		if (output) {
+			// 'h - 2' because we have two new lines in r_cons_printf
+			if (!show_vars) {
+				char *out = r_str_ansi_crop(output, 0, 0, cols, h - 2);
+				r_cons_printf("\n%s\n", out);
```
Comment (radare) [inline libr/core/vmenus.c]: Missing spacr

File: libr/core/vmenus.c [hunk 8]
```diff
@@ -2968,7 +2989,19 @@ static ut64 r_core_visual_anal_refresh (RCore *core) {
...
+				free(out);
+				R_FREE (output);
+			} else {
+				r_cons_printf("\n%s\n", output);
```
Comment (radare) [inline libr/core/vmenus.c]: M Issing space

File: libr/core/vmenus.c [hunk 9]
```diff
@@ -29,6 +29,8 @@ typedef struct {
 	const char *optword;
 } RCoreVisualTypes;
 
+static bool show_vars = false;
```
Comment (radare) [inline libr/core/vmenus.c]: Globals arr bad

**Comment by radare** [comment]
ping @deepakchethan

**Comment by deepakchethan** [comment]
@radare ping reply.

I have made the changes now

**Pull request merged**